src/logic/map_objects/tribes/productionsite.cc

```diff
@@ -627,7 +627,13 @@ std::pair<int32_t, int32_t> ProductionSite::find_worker(OPtr<Worker> w) {
  */
 void ProductionSite::try_replace_worker(const Game* game,
                                         DescriptionIndex worker_index,
-                                        WorkingPosition* wp) {
+                                        WorkingPosition* wp,
+                                        size_t iteration) {
+	if (iteration >= working_positions_.size()) {
+		// cancel recursion, something went wrong
+		log_warn("Aborting try_replace_worker() after %" PRIuS " calls.", iteration);
```

Comment:
this should be made verbose to declutter the log for mat5ches with europeans or other future tribes having similar designed productionsites